### PR TITLE
Exclude the upgrade feature if the OS does not support it 

### PR DESCRIPTION
**Pull request by Tratcher**
#427

### CODE REVIEW

File: test/Microsoft.AspNetCore.Server.HttpSys.FunctionalTests/OpaqueUpgradeTests.cs [hunk 1]
```diff
@@ -17,6 +17,38 @@ namespace Microsoft.AspNetCore.Server.HttpSys
 {
     public class OpaqueUpgradeTests
     {
+        [ConditionalFact]
+        public async Task OpaqueUpgrade_DownLevel_FeatureIsAbsent()
+        {
+            if (ComNetOS.IsWin8orLater)
+            {
+                // Skip, this test is for Win7 / 2008R2
```
Comment (anurse) [inline test/Microsoft.AspNetCore.Server.HttpSys.FunctionalTests/OpaqueUpgradeTests.cs]: Should you use the [`OSSkipConditionalAttribute`](https://github.com/aspnet/Testing/blob/dev/src/Microsoft.AspNetCore.Testing/xunit/OSSkipConditionAttribute.cs) ?

Comment (Tratcher) [reply test/Microsoft.AspNetCore.Server.HttpSys.FunctionalTests/OpaqueUpgradeTests.cs]: I actually need the opposite, skip except for a specific OS. To use OSSkipConditionalAttribute I'd have to enumerate all possible windows versions, and we don't even have constants for them, we only ever use it to skip Win7/R2.

Comment (anurse) [reply test/Microsoft.AspNetCore.Server.HttpSys.FunctionalTests/OpaqueUpgradeTests.cs]: Right. Test Conditions are pretty easy to write though, maybe just write a quick one in this repo to do that? Then it gets properly "Skipped" with a justification message.

Comment (anurse) [reply test/Microsoft.AspNetCore.Server.HttpSys.FunctionalTests/OpaqueUpgradeTests.cs]: Just copy-paste-invert the existing one even ;)

**Comment by jkotalik** [comment]
Test failed on AppVeyor: Microsoft.AspNetCore.Server.HttpSys.Listener.RequestTests.Request_OverlongUTF8Path(requestPath: "/%F0%80%BF%BF", expectedPath: "/%F0%80%BF%BF")
 System.TimeoutException : AcceptAsync has timed out.

**Pull request merged**